server/plugin/plugin.go

```diff
@@ -73,10 +74,22 @@ func (p *Plugin) OnActivate() error {
 	}
 
 	command.Register(p.API.RegisterCommand)
+
+	p.telemetryClient, err = telemetry.NewRudderClient()
+	if err != nil {
+		p.env.bot.Errorf("Cannot create telemetry client. err=%v", err)
```

Comment:
If this fails. `p.telemetryClient` is `nil`. I'm not sure if it's worth to care about the complexity. 1/5 to just return with an error here.

Reply:
I think we do not want the plugin not  to activate due to an outage of Rudder servers, or the plugin being isolated of the internet, or a firewall blocking non-api calls. If we were to return with an error here, cases like the ones I just said would make the plugin unusable.

Reply:
Good point, makes sense :+1: Does the code need an update to handle that case?

Reply:
No, telemetryClient will be set to nil, and a when used to create a new tracker, a tracker with a nil client will just not track.

Reply:
We could make that more obvious with a check like this:
```
if p.telemetryClient == nil {
   telemetry.NewTracker(nil, ....)
} else {
   telemetry.NewTracker(p.telemetryClient, ....)
}
```
But I think this is not the way to go.